parser.go

```diff
@@ -407,6 +411,9 @@ func (r *Rule) protocol(key item, l *lexer) error {
 	if key.typ != itemProtocol {
 		panic("item is not a protocol")
 	}
+	if !inSlice(key.value, []string{"tcp", "tcp-stream", "tcp-pkt", "udp", "icmp", "ip", "http", "ftp", "tls", "smb", "dns", "dcerpc", "ssh", "smtp", "imap", "modbus", "dnp3", "enip", "nfs", "ikev2", "krb5", "ntp", "dhcp"}) {
```

Comment:
let's put this slice in alphabetical order, which should make it easier to identify missing pieces.
Additionally, while I'm not sure all of these are live the `rust` parser folder in the main branch for Suricata shows other protocols, though not all of those are `app-layer` protos... `tftp` is though.

https://github.com/OISF/suricata/tree/master/rust/src